src/adaptors/adaptor_common.h

```diff
@@ -0,0 +1,92 @@
+#ifndef __adaptor_common_h__
+#define __adaptor_common_h__ 1
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "entity.h"
+
+#include "qpid/dispatch/dispatch.h"
+#include "qpid/dispatch/threading.h"
+#include "qpid/dispatch/atomic.h"
+#include "qpid/dispatch/log.h"
+#include "qpid/dispatch/alloc_pool.h"
+
+typedef enum {
+    VERSION_HTTP1,
+    VERSION_HTTP2,
+} qd_http_version_t;
+
+typedef enum {
+    QD_AGGREGATION_NONE,
+    QD_AGGREGATION_JSON,
+    QD_AGGREGATION_MULTIPART
+} qd_http_aggregation_t;
+
+// Currently we only have TCP and HTTP (HTTP/1 and HTTP/2) adaptors
+typedef enum {
+    TCP,
+    HTTP
+} qd_adaptor_type_t;
```

Comment:
Why do we separate the HTTP v1 and v2 as sub-categories of HTTP?  Aren't HTTP1 and HTTP2 completely separate adaptors?  Might it be simpler so just have HTTP1 and HTTP2 enums and remove the version?

Reply:
When it comes to handling of config and creation of listeners and connectors, the HTTP1 and the HTTP2 adaptors behave exactly the same, so at this point in time, having two adaptor types seemed to be enough, but if the need arises, we should remove HTTP and add HTTP1 and HTTP2

Reply:
I would say that TCP behaves exactly the same as well.  The event-channel stuff is optional configuration that could just as well apply to TCP.

Reply:
Originally the version enum was only part of the HTTP/* common stuff - it holds the value of the protocolVersion field in the httpConnector/httpListener schema entity (since these management entities are used for both HTTP1 and 2 - the version field is the only distinguishing element).

That said if we're introducing a new "qd_adaptor_type_t" enum then we could fold the version into it like tross suggests.